### PR TITLE
Feat/adaptive and exact dark theme

### PR DESCRIPTION
**Pull request by pan-pawel**

### CODE REVIEW

File: src/components/Appbar/AppbarHeader.tsx [hunk 1]
```diff
@@ -96,21 +91,22 @@ class AppbarHeader extends React.Component<Props> {
...
     } = this.props;
-    const { dark: isDarkTheme, colors } = rest.theme;
+    const { dark: isDarkTheme, colors, mode } = rest.theme;
     const {
       height = DEFAULT_APPBAR_HEIGHT,
-      elevation = 4,
-      backgroundColor = isDarkTheme && !primary
-        ? overlay(4, colors.surface)
-        : colors.primary,
+      elevation = 0,
```
Comment (Trancever) [inline src/components/Appbar/AppbarHeader.tsx]: Why default value is now 0 and was 4 before?

File: src/components/Appbar/AppbarHeader.tsx [hunk 2]
```diff
@@ -136,13 +132,13 @@ class AppbarHeader extends React.Component<Props> {
       >
         {/* $FlowFixMe: There seems to be conflict between Appbar's props and Header's props */}
         <Appbar
+          //@ts-ignore
```
Comment (Trancever) [inline src/components/Appbar/AppbarHeader.tsx]: Why ts ignore? :D

File: src/styles/DarkTheme.tsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { Theme } from '../types';
 const DarkTheme: Theme = {
   ...DefaultTheme,
   dark: true,
+  mode: 'adaptive',
```
Comment (Trancever) [inline src/styles/DarkTheme.tsx]: Shouldn't we also set this mode property in defaultTheme? Isn't it gonna break somewhere?

File: src/components/Appbar/Appbar.tsx [hunk 4]
```diff
@@ -136,7 +138,8 @@ class Appbar extends React.Component<Props> {
     }
     return (
       <Surface
-        style={[{ backgroundColor }, styles.appbar, restStyle]}
+        //@ts-ignore
```
[the remark this answers is not in the document]
Comment (pan-pawel) [reply src/components/Appbar/Appbar.tsx]: my mistake, fixed

File: src/components/Appbar/AppbarHeader.tsx (hunk 1 above)
Comment (pan-pawel) [reply src/components/Appbar/AppbarHeader.tsx]: my mistake, fixed

**Pull request merged**